src/main/java/com/zendesk/maxwell/MaxwellFilter.java

```diff
@@ -1,57 +1,106 @@
+ */
 public class MaxwellFilter {
-	private final HashSet<String> includeDatabases = new HashSet<>();
-	private final HashSet<String> excludeDatabases = new HashSet<>();
-	private final HashSet<String> includeTables    = new HashSet<>();
-	private final HashSet<String> excludeTables    = new HashSet<>();
+	private final ArrayList<Pattern> includeDatabases = new ArrayList<>();
+	private final ArrayList<Pattern> excludeDatabases = new ArrayList<>();
+	private final ArrayList<Pattern> includeTables = new ArrayList<>();
+	private final ArrayList<Pattern> excludeTables = new ArrayList<>();
+
 	private final HashMap<String, Integer> rowFilter = new HashMap<>();
 
-	public void includeDatabase(String dbName) {
-		throwUnlessEmpty(excludeDatabases, "database");
-		includeDatabases.add(dbName);
+	public MaxwellFilter() { }
+	public MaxwellFilter(String includeDatabases,
+						 String excludeDatabases,
+						 String includeTables,
+						 String excludeTables) throws MaxwellInvalidFilterException {
+		if ( includeDatabases != null ) {
+			for (String s : includeDatabases.split(","))
+				includeDatabase(s);
+		}
+
+		if ( excludeDatabases != null ) {
+			for (String s : excludeDatabases.split(","))
+				excludeDatabase(s);
+		}
+
+		if ( includeTables != null ) {
+			for ( String s : includeTables.split(",") )
+				includeTable(s);
+		}
+
+		if ( excludeTables != null ) {
+			for (String s : excludeTables.split(","))
+				excludeTable(s);
+		}
 	}
 
-	public void excludeDatabase(String dbName) {
-		throwUnlessEmpty(includeDatabases, "database");
-		excludeDatabases.add(dbName);
+
+	public void includeDatabase(String name) throws MaxwellInvalidFilterException {
+		includeDatabases.add(compile(name));
 	}
 
-	public void includeTable(String tblName) {
-		throwUnlessEmpty(excludeTables, "table");
-		includeTables.add(tblName);
+	public void excludeDatabase(String name) throws MaxwellInvalidFilterException {
+		excludeDatabases.add(compile(name));
 	}
 
-	public void excludeTable(String tblName) {
-		throwUnlessEmpty(includeTables, "table");
-		excludeTables.add(tblName);
+	public void includeTable(String name) throws MaxwellInvalidFilterException {
+		includeTables.add(compile(name));
 	}
 
-	public void addRowConstraint(String field, Integer value) {
-		rowFilter.put(field, value);
+	public void excludeTable(String name) throws MaxwellInvalidFilterException {
+		excludeTables.add(compile(name));
 	}
 
-	private boolean matchesDatabase(String dbName) {
-		if ( includeDatabases.size() > 0 ) {
-			return includeDatabases.contains(dbName);
+	private Pattern compile(String name) throws MaxwellInvalidFilterException {
+		if ( name.startsWith("/") ) {
+			if ( !name.endsWith("/") ) {
+				throw new MaxwellInvalidFilterException("Invalid regular expression: " + name);
+			}
+			return Pattern.compile(name.substring(1, name.length() - 1));
 		} else {
-			return !excludeDatabases.contains(dbName);
+			return Pattern.compile("^" + name.trim() + "$");
```

Comment:
er. Is this even possible? Cmd line arguments themselves are separated with spaces. The more natural assumption would be that db/table names might have spaces in them (if mysql supports that), and assume users will provide db names as --include_databases='db name','another db'

Will the option parser strip out the quotes?